python/lsst/sconsUtils/builders.py

```diff
@@ -179,40 +179,29 @@ def BuildETags(env, root=None, fileRegex=None, ignoreDirs=None):
 # -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
 
 ##
-#  @brief Remove files matching the argument list starting at dir
+#  @brief Remove files matching the argument list starting at directory
 #         when scons is invoked with -c/--clean and no explicit targets are listed
 #
 #   E.g. CleanTree(r"*~ core")
 #
-#   If recurse is True, recursively descend the file system; if
-#   verbose is True, print each filename after deleting it
+#   dirPatterns allows the specification of directories to be removed
+#   If verbose is True, print each filename after deleting it
 ##
 @memberOf(SConsEnvironment)
-def CleanTree(self, files, dir=".", recurse=True, verbose=False):
+def CleanTree(self, filePatterns, dirPatterns="", directory=".", verbose=False):
     #
     # Generate command that we may want to execute
     #
-    files_expr = ""
-    for file in SCons.Script.Split(files):
-        if files_expr:
-            files_expr += " -o "
+    filesExpr = ""
+    for filePattern in SCons.Script.Split(filePatterns):
+        if filesExpr:
+            filesExpr += " -o "
 
         # quote unquoted * and [
-        files_expr += "-name %s -prune" % re.sub(r"(^|[^\\])([\[*])", r"\1\\\2", file)
-    #
-    # don't use xargs --- who knows what needs quoting?
-    #
-    action = "find %s" % dir
-    action += r" \( -name .svn -prune -o -name \* \) "
-    if not recurse:
-        action += " ! -name . -prune"
-
-    file_action = "rm -f"
-    if recurse:
-        file_action += "r"
-
-    action += r" \( %s \) -exec %s {} \;" % \
-        (files_expr, file_action)
+        filesExpr += "-name %s" % re.sub(r"(^|[^\\])([\[*])", r"\1\\\2", filePattern)
+    action = "find %s" % directory
+    action += r" \( -name .svn -prune -o -name .git -prune -o -name \* \) "
```

Comment:
There's a lot of duplication in these two cases. Same pruning, same regex substitution. Can there be a little helper function that you can create here to reduce code duplication?